chia/wallet/wallet_state_manager.py

```diff
@@ -637,13 +638,18 @@ async def new_coin_state(
                 self.log.info(f"No wallet for coin state: {coin_state}")
                 continue
 
-            if wallet_id in all_txs_per_wallet:
-                all_txs = all_txs_per_wallet[wallet_id]
-            else:
-                all_txs = await self.tx_store.get_all_transactions_for_wallet(wallet_id)
-                all_txs_per_wallet[wallet_id] = all_txs
-
-            all_outgoing = [tx for tx in all_txs if "OUTGOING" in TransactionType(tx.type).name]
+            if wallet_id not in all_txs_per_wallet:
+                all_outgoing_trades: List[TransactionRecord] = await self.tx_store.get_all_transactions_for_wallet(
+                    wallet_id, type=TransactionType.OUTGOING_TRADE
+                )
+                all_incoming_trades: List[TransactionRecord] = await self.tx_store.get_all_transactions_for_wallet(
+                    wallet_id, type=TransactionType.INCOMING_TRADE
+                )
+                all_outgoing_std_tx: List[TransactionRecord] = await self.tx_store.get_all_transactions_for_wallet(
+                    wallet_id, type=TransactionType.OUTGOING_TX
+                )
+                all_txs_per_wallet[wallet_id] = all_outgoing_std_tx + all_outgoing_trades + all_incoming_trades
+            all_outgoing_tx: List[TransactionRecord] = all_txs_per_wallet[wallet_id]
```

Comment:
`all_outgoing` used to only contain outgoing transactions, but if I'm reading this right, it will now also contain incoming trades and outgoing trades.

Was this change deliberate? if so, should the variable still be called `all_outgoing_tx`?

Reply:
I see, so this list (`all_outgoing`) is a lot bigger prior to your patch. Those 3 DB lookups are small and that's the bulk of the gain.